L1Trigger/DTTriggerPhase2/plugins/DTTrigPhase2Prod.cc

```diff
@@ -328,179 +364,460 @@ void DTTrigPhase2Prod::produce(Event& iEvent, const EventSetup& iEventSetup) {
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             -1,
+                                                                             inMPath->primitive(0)->channelId(),
+                                                                             inMPath->primitive(0)->tdcTimeStamp(),
+                                                                             latcomb[0],
+                                                                             inMPath->primitive(1)->channelId(),
+                                                                             inMPath->primitive(1)->tdcTimeStamp(),
+                                                                             latcomb[1],
+                                                                             inMPath->primitive(2)->channelId(),
+                                                                             inMPath->primitive(2)->tdcTimeStamp(),
+                                                                             latcomb[2],
+                                                                             inMPath->primitive(3)->channelId(),
+                                                                             inMPath->primitive(3)->tdcTimeStamp(),
+                                                                             latcomb[3],
+                                                                             -1}));
```

Comment:
I'm not sure I'm fully understanding this `metaPrimitive` if most of it's fields are defaulted or nulled when made like this? Why is it better as this object instead of different objects with some base type?

Reply:
This is a generic object which covers all the cases of (proto)trigger primitives: those with hit in SL1 only, those with hit in SL3 only, and those with hits in SL1 AND SL3. 
It is true that the first 9 numbers are always -1 due to the protype character of these primitives at this stage, but they are filled completely later in the code (Lines 738 and onwards), so that smae structure can be used